onadata/apps/api/viewsets/dataview_viewset.py

```diff
@@ -155,6 +155,12 @@ def get_serializer_class(self):
 
         return serializer_class
 
+    def filter_queryset(self, queryset):
+        """
+        Remove deleted DataViews
+        """
+        return super().filter_queryset(queryset.filter(deleted_at=None))
```

Comment:
@FrankApiyo I think this is best handled when defining the `queryset` attribute [here](https://github.com/onaio/onadata/pull/2721/files#diff-fe902cbda39612b13651606735c1974aea1e9e919b9c8b0f419589ed9af07d8fR129)